### PR TITLE
refactor(diagnostic): move diagnostic UX out of core nono crate

### LINKED ISSUE #1018: Move diagnostic UX out of core nono crate
## Problem

`crates/nono/src/diagnostic.rs` currently has residues, where it mixes core sandbox logic with CLI/product UX. That weakens the intended boundary where the core `nono` crate is a pure sandbox primitiveand `nono-cli` owns policy, profiles, hooks, and user-facing output.

Some parts do seem core-appropriate and can remain.

- `DenialRecord`
- `DenialReason`
- `SandboxViolation`
- Access-mode mapping for native sandbox events

But `DiagnosticFormatter` should be CLI-owned. It renders user-facing footer text, suggests CLI flags, references `nono learn`, references `nono why`, describes profile bypass
behavior, and embeds policy/group explanation concepts.

## Evidence

Examples from `crates/nono/src/diagnostic.rs`:

- `PolicyExplanation` is tied to `nono why`, group names, policy details, and suggested CLI flags.
- `DiagnosticFormatter` renders human-facing diagnostic footer text.
- Guidance includes `nono learn -- ...` and `nono why --path ...`.
- Fix suggestions include CLI flags such as `--read`, `--write`, `--allow`, `--allow-net`, and Unix socket grant flags.
- Some guidance references user profiles and `filesystem.bypass_protection`.

The CLI already assembles the context and prints the footer from `crates/nono-cli/src/exec_strategy.rs`, so the ownership naturally appears to sit there.

## Suggested Direction

Split the module along the crate boundary:

### Keep in core `nono`

Structured sandbox facts and reusable primitives:

- Denial/event record types
- Native sandbox violation records
- Access-mode mapping helpers
- Possibly neutral stderr/path observation types if they are intended as library API

### Move to `nono-cli`

User-facing diagnostic behavior:

- `DiagnosticFormatter`
- Footer wording
- CLI flag suggestions
- `nono learn` / `nono why` guidance
- Profile-save and bypass guidance
- Policy explanation rendering
- Human-oriented stderr heuristics, if these are primarily for CLI UX

### PR DESCRIPTION
**Pull request by SequeI**
## Linked Issue

Closes #1018

## Summary

Relocate DiagnosticFormatter and all user-facing diagnostic rendering from the core 'nono' crate to 'nono-cli', restoring the library/CLI boundary documented in AGENTS.md: the library stays a pure sandbox primitive and the CLI owns policy, footer text, flag suggestions, and stderr heuristics.

Core 'nono::diagnostic' now exposes only structured, policy-free facts: DenialReason, DenialRecord, IpcDenialRecord, SandboxViolation, and the seatbelt_operation_to_access mapping. Everything else (DiagnosticFormatter, PolicyExplanation, ErrorObservation, analyze_error_output, and the footer rendering with 'nono why'/'nono learn' guidance and CLI flag suggestions) moves to crates/nono-cli/src/diagnostic/.

The bulk of this change is a near-verbatim move; the only behavioral edits:

- Remove dead code surfaced by the move (previously masked by pub visibility in the library): PolicyExplanation.details/policy_source fields (written but never read), and the unused DiagnosticFormatter methods with_blocked_protected_file, detect_protected_file_in_error, and format_summary.
- Deduplicate suggested_flag_parts: the formatter now uses the single query_ext implementation, which guards against suggesting access to the root filesystem. This fixes an edge case where a nonexistent path directly under '/' could yield a '--allow /' suggestion.

No change to diagnostic footer output in normal cases; the moved unit tests are the rendering contract and pass unchanged.

## Checklist

- [x] An issue exists and is linked above
- [x] All commits are signed-off, using [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin)
- [x] All new code follows the project's coding standards ([CLAUDE.md](CLAUDE.md)) and is covered by tests
- [ ] Public-facing changes are paired with documentation updates
- [ ] Release note has been added to `CHANGELOG.md` if needed

### CODE REVIEW

File: crates/nono-cli/src/exec_strategy.rs [hunk 1]
```diff
@@ -1475,17 +1475,13 @@ fn build_policy_explanations(
         match crate::query_ext::query_path(&path, access, caps, &[]) {
```
[the remark this answers is not in the document]
Comment (SequeI) [reply crates/nono-cli/src/exec_strategy.rs:1475]: preexisting, out of scope for this pr imo

**Review by lukehinds — APPROVED** [review]
